### PR TITLE
Purge AppearanceVisualizer

### PR DESCRIPTION
**Pull request by TemporalOroboros**
## About the PR

Removes visualizer implementations

### TODO: 
- [ ] Fix _everything_

**Media**

- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

### CODE REVIEW

**Comment by mirrorcult** [comment]
the PR of all time

**Comment by TemporalOroboros** [comment]
Currently airlocks and lockers are broken. It will take a while to fix.

**Review by AJCM-git — CHANGES_REQUESTED** [review]
You are my hero
Separate the components/systems to their own files, use the `Access` attribute above the class instead of in every field since im pretty sure it inherits to them

**Comment by EmoGarbage404** [comment]
Please just break up this pr before it dies to endless conflicts.

File: Content.Client/Chemistry/Visualizers/FoamVisualizer.cs [hunk 1]
```diff
@@ -1,68 +1,66 @@
...
 
-        [DataField("animationState")]
-        private string _state = "foam-dissolve";
+    private void OnInit(EntityUid uid, FoamVisualizerComponent comp, ComponentInit args)
+    {
+        var flick = new AnimationTrackSpriteFlick();
+        flick.LayerKey = FoamVisualLayers.Base;
+        flick.KeyFrames.Add(new AnimationTrackSpriteFlick.KeyFrame(comp.State, 0f));
+        comp.Animation = new Animation {Length = TimeSpan.FromSeconds(comp.Delay)};
+        comp.Animation.AnimationTracks.Add(flick);
```
Comment (AJCM-git) [inline Content.Client/Chemistry/Visualizers/FoamVisualizer.cs:40]: Make sure this and every animation are a single object initializer so they are easier to understand

File: Content.Client/Disposal/Visualizers/DisposalUnitVisualizer.cs [hunk 2]
```diff
@@ -5,165 +5,160 @@
...
+        sound.KeyFrames.Add(new AnimationTrackPlaySound.KeyFrame(comp.FlushSound.GetSound(), 0));
+    }
 
-        private void ChangeState(AppearanceComponent appearance)
+    protected override void OnAppearanceChange(EntityUid uid, DisposalUnitVisualizerComponent comp, ref AppearanceChangeEvent args)
+    {
+        var sprite = args.Sprite;
+        if (sprite == null)
+            return;
+        if (!AppearanceSystem.TryGetData(uid, Visuals.VisualState, out VisualState state, args.Component))
```
Comment (AJCM-git) [inline Content.Client/Disposal/Visualizers/DisposalUnitVisualizer.cs:91]: make sure to always use the generic version `TryGetData<>()` so the cast is done in the method

**Review by AJCM-git — CHANGES_REQUESTED** [review]

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
This already conflicts with open prs (e.g. disposalunits) + needs a lot of reviewing (visualizers being used where components already exist) so please split it.

**Comment by TemporalOroboros** [comment]
> This already conflicts with open prs (e.g. disposalunits) + needs a lot of reviewing (visualizers being used where components already exist) so please split it.

Aight, 36 prs incoming.

**Comment by TemporalOroboros** [comment]
Atomized

**Pull request closed without merging**